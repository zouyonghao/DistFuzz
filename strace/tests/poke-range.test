#!/bin/sh -efu
#
# Check poke injection argument range.
#
# Copyright (c) 2021 The strace developers.
# All rights reserved.
#
# SPDX-License-Identifier: GPL-2.0-or-later

. "${srcdir=.}/init.sh"

run_prog ../chdir > /dev/null
run_strace -a10 -e trace=chdir \
	   -einject=chdir:when=2+:poke_enter=@arg2=5374726163653f00 \
	   $args >"$EXP" 2>"$LOG.err"

match_diff "$LOG" "$EXP"

match_grep "$LOG.err" "$srcdir/poke-range.expected"
