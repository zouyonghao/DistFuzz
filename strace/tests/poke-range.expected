.*: Failed to tamper with process [[:digit:]]+: requested to tamper with argument #2, but system call 'chdir' has only 1 arguments
